packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -659,8 +703,19 @@ export function checkIfContextChanged(
       ? context._currentValue
       : context._currentValue2;
     const oldValue = dependency.memoizedValue;
-    if (!is(newValue, oldValue)) {
-      return true;
+    if (enableContextProfiling && dependency.compare != null) {
```

Comment:
This is where my other comment re: expecting the compare to always be present comes in. Seems potentially if you go from no compare to compare and vice versa since you switch to doing `is` based comparison and ignoring the last compared value